pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -346,7 +379,9 @@ def hpc_load(self, checkpoint_path: str, on_gpu: bool):
             model.cuda(self.trainer.root_gpu)
 
         # restore training state
-        self.restore_training_state(checkpoint)
+        self._loaded_checkpoint = checkpoint
+        self.restore_training_state()
+        self._loaded_checkpoint = dict()
```

Comment:
The fact that this is setting/unsetting is necessary makes me thing the checkpoint should be a method parameter

Reply:
good observation. had to put this here but it will disappear in the follow up pr, where hpc_load will get merged with the regular restore. 

hpc load does the same thing as the regular restore except call `on_hpc_load` instead of `on_load_checkpoint`. 
so I will be able to remove this code duplication. 

I can add a fixme comment if you agree?

Reply:
actually nevermind, reverted back so no need to set and reset this variable.